File: src/main/java/io/bindernews/thegrackle/cards/CrashLanding.java
package io.bindernews.thegrackle.cards;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.common.DamageAction;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.stances.NeutralStance;
import io.bindernews.bnsts.CardVariables;
import io.bindernews.thegrackle.cardmods.RequireAloftMod;
import lombok.val;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class CrashLanding extends BaseCard {
    public static final CardConfig C =
            new CardConfig("CrashLanding", CardType.ATTACK, CardRarity.BASIC, CardTarget.ENEMY);
    static final CardVariables VARS = CardVariables.config(c -> {
        c.cost(1, -1);
        c.damage(12, 20);
        c.addModifier(RequireAloftMod::new);
    });

    public CrashLanding() {
        super(C, VARS);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        val fx = AbstractGameAction.AttackEffect.SLASH_DIAGONAL;
        addToBot(new DamageAction(m, new DamageInfo(p, damage, damageType), fx));
        addToBot(iop().changeStance(p, NeutralStance.STANCE_ID));
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/CrashLanding.kt
package io.bindernews.thegrackle.cards

import com.megacrit.cardcrawl.actions.AbstractGameAction.AttackEffect
import com.megacrit.cardcrawl.actions.common.DamageAction
import com.megacrit.cardcrawl.cards.DamageInfo
import com.megacrit.cardcrawl.core.AbstractCreature
import com.megacrit.cardcrawl.stances.NeutralStance
import io.bindernews.bnsts.CardVariables
import io.bindernews.thegrackle.cardmods.AloftDmgUnaffectedMod
import io.bindernews.thegrackle.cardmods.RequireAloftMod
import io.bindernews.thegrackle.helper.ModInterop.Companion.iop

class CrashLanding : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        val fx = AttackEffect.SLASH_DIAGONAL
        addToBot(DamageAction(m, DamageInfo(p, damage, damageType), fx))
        addToBot(iop().changeStance(p, NeutralStance.STANCE_ID))
    }

    companion object {
        @JvmField
        val C = CardConfig("CrashLanding", CardType.ATTACK, CardRarity.BASIC, CardTarget.ENEMY)
        val VARS = CardVariables().apply {
            cost(1, -1)
            damage(8, 12)
            addModifier(RequireAloftMod())
            addModifier(AloftDmgUnaffectedMod())
        }
    }
}